aten/src/ATen/native/cpu/UpSampleKernelAVXAntialias.h

```diff
@@ -223,24 +240,27 @@ void ImagingResampleVertical(
   const int64_t* idx_ptr_xmin = vert_indices_weights[0].data_ptr<int64_t>();
   const int64_t* idx_ptr_size = vert_indices_weights[1].data_ptr<int64_t>();
 
-  uint32_t* unpacked_output_p = (uint32_t*) unpacked_output.data_ptr<uint8_t>();
-  const uint32_t* unpacked_input_p = (uint32_t*) unpacked_input.data_ptr<uint8_t>();
+  uint8_t* unpacked_output_p = unpacked_output.data_ptr<uint8_t>();
+  const uint8_t* unpacked_input_p = unpacked_input.data_ptr<uint8_t>();
```

Comment:
Curious why these are converted to `uint8` pointers. Is it because there was no "good" reason for them to be uint32 in the first place?

Reply:
We are dealing in `upsample_avx_bilinear_uint8` with tensors of dtype `uint8` and num channels = 1, 2, 3, 4, channels first/last. Previously, we copied input tensors into num channels 4, channels last if the input is not like this. So, `ImagingResampleVertical` and `ImagingResampleHorizontal` both could extract data as uint32 which corresponds to 4 uint8 values = RGBA. It was OK to deal with uint32 data and sometimes even better as AVX ops like `mm_cvtepu8_epi32` requires int32 memory alignment.

In this PR, we want to avoid copying input tensor with num channels 3, channels last into RGBA to speed-up the processing, so if we fetch data as uint32 we'll get wrong strides: (r1, g1, b1, r2), (g2, b2, r3, g3), ...